backend/scripts/rerun_failing_test_executions.py

```diff
@@ -0,0 +1,87 @@
+# ruff: noqa: T201
+
+import argparse
+import re
+
+import requests
+
+TO_API_URL = "https://test-observer-api.canonical.com"
+
+
+def main(artefact_id: int, test_case_regex: str, reversed: bool):
+    artefact_builds = requests.get(
+        f"{TO_API_URL}/v1/artefacts/{artefact_id}/builds"
+    ).json()
+
+    test_case_matcher = re.compile(test_case_regex)
+
+    relevant_test_executions = [
+        te
+        for ab in artefact_builds
+        for te in ab["test_executions"]
+        if te["review_decision"] == []
+        and not te["is_rerun_requested"]
+        and te["status"] == "FAILED"
+    ]
+
+    test_execution_ids_to_rerun = []
+    for te in relevant_test_executions:
+        test_results = requests.get(
+            f"{TO_API_URL}/v1/test-executions/{te['id']}/test-results"
+        ).json()
+
+        matching_failed_tests = (
+            tr
+            for tr in test_results
+            if tr["status"] == "FAILED" and test_case_matcher.match(tr["name"])
+        )
+
+        first_failing_test = next(matching_failed_tests, None)
+
+        if first_failing_test and not reversed:
+            test_execution_ids_to_rerun.append(te["id"])
+            print(
+                f"will rerun {te['environment']['name']}"
+                f" for failing {first_failing_test['name']}"
+            )
+        elif not first_failing_test and reversed:
+            test_execution_ids_to_rerun.append(te["id"])
+            print(
+                f"will rerun {te['environment']['name']}"
+                f" as no failing matches found and reversed option is set"
+            )
+
+    should_rerun = (
+        input(
+            f"Will rerun {len(test_execution_ids_to_rerun)}"
+            " test executions is that ok? (Y/n) "
+        )
+        == "Y"
```

Comment:
“y” should also be an acceptable value, and a simple “enter”
When you specify `Y/n`, the uppercase choice means a default option (probably you already knew that). 

If you want to deal with exact value, it would be more user-friendly to specify it in the following way:

```suggestion
            " test executions is that ok? (yes/no) "
        )
        == "yes"
```

Reply:
Oh I didn't know about the uppercase convention